### PR TITLE
Issue #49 and kill a few more pylint warnings

### PR DESCRIPTION
**Pull request by sberl**
Both data_path and local_tmp must be readable and writeable folders. 
Also clean a few more lint errors.

### CODE REVIEW

**Comment by sberl** [comment]
Looking at the pylon output for config.py I don't understand how to fix the error on line 226.

config.py:226:20: R1705: Unnecessary "else" after "return" (no-else-return)

Any thoughts?

File: supersid/config.py [hunk 1]
```diff
@@ -410,35 +422,31 @@ def supersid_check(self):
...
-    cfg = Config(cfg_filename)
-    cfg.supersid_check()
-    if cfg.config_ok:
-        assert(1 == len(cfg.filenames))
-        print("Config file '{}' read successfully".format(cfg.filenames[0]))
+    """Read and return the configuration or terminate the program."""
+    config = Config(cfg_filename)
+    config.supersid_check()
+    if config.config_ok:
+        assert len(cfg.filenames) == 1
```
Comment (fenrog) [inline supersid/config.py:429]: `assert len(cfg.filenames) == 1`
should read as
`assert len(config.filenames) == 1`

**Review by fenrog — CHANGES_REQUESTED** [review]

File: supersid/config.py (hunk 1 above)
Comment (sberl) [reply supersid/config.py:429]: Good catch

**Comment by fenrog** [comment]
In order to resolve
>config.py:226:20: R1705: Unnecessary "else" after "return" (no-else-return)

this code
```
                    if pdefault is None:  # missing mandatory parameter
                        self.config_ok = False
                        self.config_err = "'"+pkey+"' is not found in '%s'. " \
                            "Please check." % filename
                        return
                    else:  # optional, assign default
                        self.setdefault(pkey, pdefault)
```
should read as
```
                    if pdefault is None:  # missing mandatory parameter
                        self.config_ok = False
                        self.config_err = "'"+pkey+"' is not found in '%s'. " \
                            "Please check." % filename
                        return
                    self.setdefault(pkey, pdefault)
```

**Comment by sberl** [comment]
Yes. Of course. Now it makes perfect sense. -Steve

**Pull request merged**